3495/0/triangle.rb

```diff
@@ -0,0 +1,74 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+
+# triangle Project Code.
+
+# triangle analyzes the lengths of the sides of a triangle
+# (represented by a, b and c) and returns the type of triangle.
+#
+# It returns:
+#   :equilateral  if all sides are equal
+#   :isosceles    if exactly 2 sides are equal
+#   :scalene      if no sides are equal
+#
+# The tests for this method can be found in
+#   about_triangle_project.rb
+# and
+#   about_triangle_project_2.rb
+#
+# :reek:UtilityFunction
+class Triangle
+  attr_reader :side_a, :side_b, :side_c
+
+  def initialize(side_a, side_b, side_c)
+    @side_a = side_a
+    @side_b = side_b
+    @side_c = side_c
+  end
+
+  def type
+    if check_triangle_existence || check_triangle_existence_more ||
```

Comment:
Do you think `check_triangle_existence_more` and `check_triangle_existence_and_more` are good names for methods? I meant move whole logic to method something like `check_triangle_existence`
It may looks like
```
def type
  check_triangle_existence
  if a_equilateral?
    :equilateral
  elsif a_isosceles?
    :isosceles
  elsif a_scalene?
    :scalene
  end
end

def check_triangle_existence
  # check here everything you want and raise exception if need
end
```
It is not ideal solution, but we haven't more time to make better.
BTW all methods except `type` and `initialize` should be private.